### PR TITLE
Thread BR commissioning support

### PR DESCRIPTION
**Pull request by artokin**
Allow application to bypass Thread static configuration to allow
Border Router commissioning. Flag THREAD_USE_STATIC_LINK_CONFIG
controls usage of static link configuration settings.

### CODE REVIEW

**Comment by artokin** [comment]
@terhei , @mikter , @karsev would you please review?

File: source/borderrouter_thread_tasklet.c [hunk 1]
```diff
@@ -147,10 +149,14 @@ static int thread_interface_up(void)
     device_config.PSKd_len = len;
     memset(device_config.PSKd_ptr, 0, len + 1);
     memcpy(device_config.PSKd_ptr, param, len);
+    rf_read_mac_address(rf_mac);
+    memcpy(device_config.eui64, rf_mac, 8);
 
-    thread_link_configuration_get(&link_setup);
+    channel_list.channel_mask[0] = MBED_CONF_APP_RF_CHANNEL_MASK;
```
Comment (mikter) [inline source/borderrouter_thread_tasklet.c]: Channel list should be NULL
The correct place for these settings are in the active configuration and stack should but default to all.

This entire parameter is questionable in node init and should be deprecated

File: source/borderrouter_thread_tasklet.c [hunk 2]
```diff
@@ -133,7 +130,9 @@ static int thread_interface_up(void)
 {
     int32_t val;
     device_configuration_s device_config;
+    uint8_t rf_mac[8] = {0};
```
Comment (mikter) [inline source/borderrouter_thread_tasklet.c]: no need to initialise as there is always write

File: source/borderrouter_thread_tasklet.c [hunk 3]
```diff
@@ -147,10 +146,11 @@ static int thread_interface_up(void)
     device_config.PSKd_len = len;
     memset(device_config.PSKd_ptr, 0, len + 1);
     memcpy(device_config.PSKd_ptr, param, len);
+    rf_read_mac_address(rf_mac);
+    memcpy(device_config.eui64, rf_mac, 8);
```
Comment (mikter) [inline source/borderrouter_thread_tasklet.c]: Stack reads the eui64 from radio if initialised for 0 is this value not initialised properly?

**Pull request merged**